osu.Game.Tests/Beatmaps/Formats/LegacyBeatmapEncoderTest.cs

```diff
@@ -19,49 +22,115 @@
-            foreach (var g in beatmap.ControlPointInfo.Groups)
+            foreach (var g in tuple.beatmap.ControlPointInfo.Groups)
             {
                 ArrayList.Adapter((IList)g.ControlPoints).Sort(
                     Comparer<ControlPoint>.Create((c1, c2) => string.Compare(c1.GetType().ToString(), c2.GetType().ToString(), StringComparison.Ordinal)));
             }
         }
 
-        private IBeatmap decodeFromLegacy(Stream stream)
+        private (IBeatmap beatmap, TestLegacySkin beatmapSkin) decodeFromLegacy(Stream stream, string name)
         {
             using (var reader = new LineBufferedReader(stream))
-                return convert(new LegacyBeatmapDecoder { ApplyOffsets = false }.Decode(reader));
+            {
+                var beatmap = new LegacyBeatmapDecoder { ApplyOffsets = false }.Decode(reader);
+
+                using (var rs = new MemoryBeatmapResourceStore(stream, name))
+                {
+                    var beatmapSkin = new TestLegacySkin(beatmap, rs, name);
+                    return (convert(beatmap), beatmapSkin);
+                }
+            }
+        }
+
+        private class MemoryBeatmapResourceStore : IResourceStore<byte[]>
+        {
+            private readonly Stream beatmapData;
+            private readonly string beatmapName;
+
+            public MemoryBeatmapResourceStore(Stream beatmapData, string beatmapName)
+            {
+                this.beatmapData = beatmapData;
+                this.beatmapName = beatmapName;
+            }
+
+            public void Dispose() => beatmapData.Dispose();
+
+            public byte[] Get(string name)
+            {
+                if (name != beatmapName)
+                    return null;
+
+                byte[] buffer = new byte[beatmapData.Length];
+                beatmapData.Read(buffer, 0, buffer.Length);
+                return buffer;
+            }
+
+            public async Task<byte[]> GetAsync(string name)
+            {
+                if (name != beatmapName)
+                    return null;
+
+                byte[] buffer = new byte[beatmapData.Length];
+                await beatmapData.ReadAsync(buffer.AsMemory());
+                return buffer;
+            }
+
+            public Stream GetStream(string name)
+            {
+                if (name != beatmapName)
+                    return null;
+
+                beatmapData.Seek(0, SeekOrigin.Begin);
+                return beatmapData;
+            }
+
+            public IEnumerable<string> GetAvailableResources() => beatmapName.Yield();
+        }
+
+        private class TestLegacySkin : LegacySkin, IBeatmapSkin
+        {
+            public TestLegacySkin(Beatmap beatmap, IResourceStore<byte[]> storage, string fileName)
+                : base(new SkinInfo { Name = "Test Skin", Creator = "Craftplacer" }, storage, null, fileName)
+            {
+            }
```

Comment:
is this all really necessary? can't you just `new LegacyBeatmapDecoder().Decode(stream)`?